docs/design/package-manager-template.md

```diff
@@ -0,0 +1,218 @@
+
+- **Native checksum support**: Whether the package manager provides checksums
+- **Checksum algorithms**: Which hashing algorithms are used (SHA-256, SHA-1, etc.)
+- **Checksum sources**: Where checksums are obtained (registry metadata, computed locally, etc.)
+- **Missing checksum handling**: What happens when checksums are unavailable?
+
+### Fetching Content
+
+Describe how Hermeto should fetch dependencies on the dependency list. This will form the core of
+the `fetch-deps` command implementation.
+
+_Note: The subsections below are not required, but serve as a useful starting point_.
+
+#### Native vs. Hermeto Fetch
+
+Decide if the package manager can be trusted to fetch dependencies, or if Hermeto should "reverse
+engineer" the dependency download process:
+
+- Does the package manager have mechanisms to resolve dependencies from a fixed list?
+- Does the package manager have plugins, hooks, or other mechanisms that allow arbitrary code to be
+  executed during the download/resolution phase?
+
+In general, Hermeto should be responsible for downloading dependencies.
+
+#### Project Structure
+
+Provide directory tree diagrams of the following:
+
+- The developer's project (where dependencies are typically declared).
+- Any "cache" directories where dependencies are installed locally to disk.
+
+#### File Formats and Metadata
+
+Document any specific file format requirements:
+
+- **Package file formats**: Expected formats for downloaded packages
+- **Metadata requirements**: Additional metadata files Hermeto must provide
+- **Naming conventions**: Required naming patterns for files and directories
+- **Version handling**: How different versions should be organized
+
+#### Network Requirements
+
+Describe network-related considerations:
+
+- **Registry endpoints**: URLs and APIs Hermeto needs to access
+- **Authentication**: Any authentication requirements for package registries
+- **Rate limiting**: Considerations for API rate limits
+- **Mirror support**: Support for alternative registries or mirrors
+
+### Build Environment Config
+
+Describe how the build environment should be configured to use Hermeto's pre-fetched dependencies.
+This section will form the basis of the `generate-env` and `inject-files` commands.
+
+#### Environment Variables
+
+Describe any environment variables that need to be set so that the package manager uses the
+dependencies pre-fetched by Hermeto. A table is usually sufficient:
+
+| Variable Name | Purpose | Example Value | Required |
+|---------------|---------|---------------|----------|
+| `EXAMPLE_VAR` | Points to dependency cache | `/path/to/hermeto-deps` | Yes |
+
+#### Configuration Files
+
+Describe any files that Hermeto should generate or provide to the package manager. This will form
+the basis of the `inject-files` implementation. A tree diagram can be helpful here:
+
+```
+hermeto-deps/
+├── [package-manager-name]/
+│   ├── metadata/
+│   ├── packages/
+│   └── manager-config.json
+```
```

Comment:
This sample tree doesn't resemble anything that we do as part of that command actually. While it will differ among backends, the common pattern worth mentioning might instead be:
```
project.git/
├── <depedencies_lockfile>  # e.g. requirements.txt, yarn.lock, packages.json <<< We may overwrite this
└── <package_manager_data>/ # e.g. .bundler, .yarn, .config directory <<< We may overwrite configuration here 
    ├── cache
    ├── config
    ├── some_dir
    └── some_file
```